test/controllers/groups_controller_test.rb

```diff
@@ -1,34 +1,135 @@
+    other_user.groups << other_group
+    get :index
+    assert_equal assigns(:groups).count, 1
+    assert_not assigns(:groups).include?(other_group)
   end
 
-  def test_update
-    get :update
-    assert_response :success
+  test "should create group" do
+    sign_in @user
+    assert_difference('Group.count') do
+      post :create, group: {name: Faker::Lorem.word, description: Faker::Lorem.sentence}
+    end
+    assert_response :redirect
+    assert_redirected_to group_path(Group.first)
+  end
+
+  test "each group should have its owner" do
+    sign_in @user
+    post :create, group: {name: Faker::Lorem.word, description: Faker::Lorem.sentence}
+    assert_equal Group.first.owner, @user
+  end
+
+  test "group should not be updated by other group's members" do
+    sign_in @user
+    group = create(:group)
+    @user.groups << group
+    other_user = create(:user, goal: create(:goal))
+    group.owner = other_user
+    other_user.groups << group
+    patch :update, :id => group.id, group: {name: 'ait'}
+    assert_response 401
+  end
+
+  test "group should only be updated by group's owner" do
+    sign_in_user_and_assign_groups
+    other_user = create(:user, goal: create(:goal))
+    other_user.groups << @group
+    patch :update, :id => @group.id, group: {name: Faker::Lorem.word}
+    assert_response :redirect
+  end
+
+  test "should not update group without any parameter" do
+    sign_in_user_and_assign_groups
+    assert_raises ActionController::ParameterMissing do
+      put :update, :id => @group.id
+    end
+  end
+
+  test "should update group when either name, description, featured is changed" do
+    sign_in_user_and_assign_groups
+    updated_name = 'group_ait'
+    patch :update, :id => @group.id, group: {name: updated_name}
+    @group.reload
+    assert_response :redirect
+    assert_equal @group.name, updated_name
   end
 
-  def test_destroy
-    get :destroy
+  test "group should not be destroyed by other group's members" do
+    sign_in @user
+    group = create(:group)
+    @user.groups << group
+    other_user = create(:user, goal: create(:goal))
+    group.owner = other_user
+    other_user.groups << group
+    delete :destroy, :id => group.id
+    assert_response 401
+  end
+
+  test "should destroy group" do
+    sign_in_user_and_assign_groups
+    assert_difference('Group.count', -1) do
+      delete :destroy, id: @group.id
+    end
+    assert_response :redirect
+    assert_redirected_to groups_path
+  end
+
+  test "should feature group only when user is admin" do
+    sign_in @user
+    group = create(:group, is_featured: true )
+    group.owner = @user
+    @user.groups << group
+    role = create(:role, :name => 'Admin')
+    @user.roles << role
+    xhr :patch, :feature, id: group.id
     assert_response :success
+    assert_template 'groups/feature'
+  end
+
+  def sign_in_user_and_assign_groups
+    sign_in @user
+    @group = create(:group)
+    @group.owner = @user
+    @user.groups << @group
   end
 
```

Comment:
Please add these test cases: 

test "group widget displays members for that month only"
test "group widget shows only last 3 months"
test "clicking on the user name in the widget shows the user profile"
test "clicking on the user name in the widget shows the user profile even if the user is not logged in"
test "clicking on the group name in the widget shows the group details"
test "clicking on the group name in the widget shows the group details even if the user is not logged in "
test "member can also embed the group widget"  // Possible feature request: We may need a "share / embed button in the widget for this itself.